demos/_includes/DemoInvoice.php

```diff
@@ -9,13 +9,23 @@
  */
 class DemoInvoice extends \atk4\data\Model
 {
+    public $dateFormat;
+
     public $title_field = 'reference';
 
     protected function init(): void
     {
         parent::init();
 
-        $this->addField('reference');
-        $this->addField('date', ['type' => 'date']);
+        $this->addField('reference', ['required' => true]);
+        $this->addField('date', [
+            'type' => 'date',
+            'required' => true,
+            'typecast' => [
+                function ($v) {
+                    return ($v instanceof \DateTime) ? date_format($v, $this->dateFormat) : $v;
```

Comment:
for what is this needed?

for UI? If so, it will affect also db typecasting, right?

Reply:
the demo use persistence array and date are not typecast without this.

Reply:
then probably use `Y-m-d` here as this is later expected once we move to DBAL

ie. no need for `dateFormat` property at all, this will never handle ui typecasting

Reply:
dateFormat is there only for Ui, i.e. to display input value according to ui_persistence and not for DB.

Reply:
I added this part as TODO to https://github.com/atk4/data/pull/690

here - ok for now